### ISSUE TITLE
Improve Printing For Acceptance Tests

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#139]
@jrperritt: I was playing around with some acceptance test changes and wanted to get your thoughts on whether they were worthwhile or not. They're broken into the top 6 commits here:

https://github.com/jtopjian/gophercloud/commits/acctest-updates

**Comment by jrperritt** [comment]
Can you summarize the changes (what you changed and why)?

**Comment by jtopjian** [comment]
Sure:

There are two changes being made:

1. Not passing the `choices` struct between the Test functions and the shared collection of resource creation functions. For example, when creating a server for testing purposes, the current way is:

```
server, err := CreateServer(t, client, choices)
```

I'm proposing to change this to:

```
server, err := CreateServer(t, client)
```

And then create `choices` within the `CreateServer` function. Since `choices` just pulls information from the environment, it's created in the same way. I think this is a good idea because it makes `CreateServer` more independent by removing required function parameters. Though I could be wrong -- thoughts?

2. Changing the way resources are printed (`Print<Resource>`) by using some type of package / function that can easily dump all attributes. This makes these functions more maintainable for added/removed fields and less prone to typos. 

Different ways of doing this were discussed a bit in #128. I'm happy to yield to whatever decision is made and alter accordingly.

**Comment by jrperritt** [comment]
> Though I could be wrong -- thoughts?

I think that's fine.

> Changing the way resources are printed (Print<Resource>) by using some type of package / function 

I'd prefer not to use a third-party package just for printing. I think a function with the following logic should work for any `struct`:
- create a var (say, `sf`) of type `[]reflect.StructField`
- for the `struct` (passed as a parameter), iterate over NumFields, stripping StructField.Tag and appending to `sf`
- create the struct type with `reflect.StructOf`, passing `sf` as argument
- print created `struct`

**Comment by jtopjian** [comment]
> I'd prefer not to use a third-party package just for printing.

I agree.

> I think a function with the following logic should work for any struct

OK, I can toy around with this.

Thanks!

**Comment by jtopjian** [comment]
After looking into this for a considerable amount of time, I've realized that implementing a pretty printer is _not_ a simple task.

Taking a step back, the goal of this would be to print a resource's details in readable form. Basically:

```go
fmt.Printf("%#v\n", securityGroup)
```

But readable.

The easiest way is to just marshal the resource into JSON, but this strips away some data: namely the absolute field name and the type of the value. 

@jrperritt suggested the use of `[]reflect.StructField` et al. That works, but the implementation becomes a little complicated with nested structs / resources (for example: security group rules that are part of security groups).

In addition, I ran into issues with types like `JSONRFC3339MilliNoZ`. While that's certainly due to my implementation, it caused me to be mindful that there are other one-off problems that could happen down the road. (I also noticed that `JSONRFC3339MilliNoZ` is not being converted to JSON correctly. Maybe this will be fixed after the `time.Time` work is done).

And in the end, this method would only be ensuring the Field names are not translated to JSON. IMO, not really worth the effort.

With all of that in mind, I think there are three options:

1. Just use `%#v` to dump the resource
2. Go with plain JSON
3. Implement a heavyweight pretty printer like `go-spew`

I'm opting for 2 for now.
